polygeist/tools/cgeist/driver.cc

```diff
@@ -1164,6 +1165,58 @@ template <typename T> static void filterFunctions(T Module) {
     Func.erase();
 }
 
+static int readHostModule(MLIRContext &context, ModuleOp module,
```

Comment:
I would make it return a bool (false indicating that the module was not read correctly).